### ISSUE TITLE
Create a Jenkins job to test launching multiple copies & update of zowe from the same runtime

### ISSUE THREAD

**Issue by stevenhorsman** [zowe/zowe-install-packaging#517]
Once the dev work in #433 is done we should be able to run multiple copies of zowe with different config from the same runtime (binaries) and also update the runtime without losing any config data (assuming that at each release any migration of config is done.

Also need to test bringing up sub-sets of components and test them

We should create pipeline jobs for these tests

**Comment by jackjia-ibm** [comment]
Extra comment from Steven Horsman  11:24 AM @ Apr 20, 2020:

So this issue was created before I separated runtime and instance, so they were all together then. There was a plan to have a migrate step for each component (still on the backlog), so they could manually migrate any changed property names, but we’d want to make sure they didn’t replace the extender added ones. I’m not sure if there is plans to create any automation to test adding any zowe extensions (like the IMS one maybe), but the scenario would now be:

- Install 1.9.0 (or a back level)
- Install extension (updating instance.env to include it)
- Update
- Check extension still works

**Comment by jackjia-ibm** [comment]
Idea to solve multiple instances in one server:

so far what i’m thinking is the difficulty of uninstall. we currently try to wipe everything possible on the server. but if the user has 2 or 3 instances on the server, we need a way to define the instance, and provide option to uninstall only one of them or all of them
we currently have all the variables defined for the “default” instance, i’m thinking to define another variable as list `extra_instances`
so any variable defined for default instance can be copied/redefined in the extra_instances
for example, we have `zowe_instance_dir: /ZOWE/tmp/.zowe` defined, this is default instance,
then we have:

```yaml
extra_instances:
  test1:
    extra_instances: /ZOWE/tmp/.
    zowe_instance_id: A
    zowe_apiml_gateway_port: 17554
    # ....more
  jack-testa:
    extra_instances: /u/jack/.zowe
    zowe_instance_id: J
    zowe_apiml_gateway_port: 22222
    # ....more
```

when we do a full uninstall, we find all the extra instances and clean up the folders
when we say just uninstall one instance, we only remove the folders/STC/datasets defined in that instance
when do install, we can say install test1 , then all variables defined in test1 will overwrite the default instance variables, so the playbooks remains same

**Comment by jackjia-ibm** [comment]
After discussed with Steve, this item will be postponed to PI3.
